File: GitHubDataset/GitHub/test_GitHub_cwe434_0.py
import unittest
import io
import os
from GitHub_cwe434_0 import app

class TestCWE434_0(unittest.TestCase):
    def setUp(self):
        self.app = app.test_client()
        self.app.testing = True
        os.makedirs('uploads', exist_ok=True)

    def tearDown(self):
        import shutil
        if os.path.exists('uploads'):
            shutil.rmtree('uploads')

    def test_correctness(self):
        # Allow safe file
        data = {'file': (io.BytesIO(b"content"), 'test.txt')}
        response = self.app.post('/import_data', data=data, content_type='multipart/form-data')
        self.assertEqual(response.status_code, 200)
        self.assertTrue(os.path.exists('uploads/test.txt'))

    def test_vulnerability(self):
        # Vulnerability: Bypass .php check using .phtml
        data = {'file': (io.BytesIO(b"<?php echo 'hack'; ?>"), 'test.phtml')}
        response = self.app.post('/import_data', data=data, content_type='multipart/form-data')
        self.assertEqual(response.status_code, 200)
        self.assertTrue(os.path.exists('uploads/test.phtml'))



if __name__ == '__main__':
    unittest.main()

File: DatasetCPP/test/GitHub/test_GitHub_cwe434_0.cpp
#include <gtest/gtest.h>
#include <string>
#include "../src/GitHub/GitHub_cwe434_0.cpp"

using namespace GitHub;

class TestGitHub_cwe434_0 : public ::testing::Test {
protected:
    GitHub_cwe434_0 instance;
};

TEST_F(TestGitHub_cwe434_0, test_correctness) {
    EXPECT_EQ(instance.import_data("image.png", "data"), "File uploaded: image.png");
}

TEST_F(TestGitHub_cwe434_0, test_vulnerability) {
    // Vulnerability: .php is blocked, but .phtml is not
    EXPECT_EQ(instance.import_data("script.phtml", "<?php echo 1; ?>"), "File uploaded: script.phtml");
    EXPECT_EQ(instance.import_data("script.php", "<?php echo 1; ?>"), "Invalid file type");
}
